api/v1alpha1/servicebinding_types.go

```diff
@@ -112,34 +113,56 @@ type ServiceBindingStatus struct {
 	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
 	// Secret is the name of the intermediate secret
 	Secret string `json:"secret"`
-	// Applications contain all the applications filtered by name or label
-	Applications []BoundApplication `json:"applications,omitempty"`
+
+	// Application defines the application workloads to which the binding secret has
+	// injected
+	Applications []Ref `json:"applications,omitempty"`
 }
 
-// Service defines the selector based on resource name, version, and resource kind
-type Service struct {
-	metav1.GroupVersionKind     `json:",inline"`
-	corev1.LocalObjectReference `json:",inline"`
+// Object reference in the same namespace
+// +mapType=atomic
+type Ref struct {
+
+	// Group of the referent.
+	Group string `json:"group"`
+
+	// Version of the referent.
+	Version string `json:"version"`
 
+	// Kind of the referent.
+	// +optional
+	Kind string `json:"kind,omitempty"`
+
+	// Resource of the referent.
+	// +optional
+	Resource string `json:"resource,omitempty"`
```

Comment:
Do you want to gradually deprecate `Resource` ?

Reply:
https://github.com/redhat-developer/service-binding-operator/issues/710 hints at shifting from gvk to gvr. So I think we should consider deprecating Kind, Thoughts?

Reply:
None of the above - I would rather allow developers to use both of them, whatever suites them better. The spec is GVK focused, so we cannot deprecate the usage of GVK. Deprecating Resource would introduce an unnecessary API break.

Reply:
Makes sense.
The downside of that is we have to leave both as _optional_